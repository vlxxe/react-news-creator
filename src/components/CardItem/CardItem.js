import React from "react"

export const CardItem = ({ card }) => {
  return (
    <div
      className="uk-card uk-card-default"
      style={{ display: "block", width: 580, marginBottom: 15 }}
      key={card.id}
    >
      <div className="uk-card-header">
        <div className="uk-grid-small uk-flex-middle" uk-grid="">
          <div className="uk-width-auto">
            <img
              className="uk-border-circle"
              width="40"
              height="40"
              src={card.profile.img}
              alt="user-pic"
            />
          </div>
          <div className="uk-width-expand">
            <h3 className="uk-card-title uk-margin-remove-bottom">
              {card.title}
            </h3>
            <div className="uk-flex uk-flex-between">
              <p className="uk-text-meta uk-margin-remove-top uk-margin-remove-bottom">
                {card.profile.name}
              </p>
              <span className="uk-text-meta">{card.date}</span>
            </div>
          </div>
        </div>
      </div>
      <div className="uk-card-body">
        <p>{card.text}</p>
      </div>
    </div>
  )
}
