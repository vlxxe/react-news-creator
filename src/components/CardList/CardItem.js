import React from "react";
import { NavLink } from "react-router-dom";
import { EditCard } from "../EditCard/EditCard";

export const CardItem = ({ card }) => {
  return (
    <div
      className="uk-card uk-card-default"
      style={{ width: 580, marginBottom: 15 }}
    >
      <div className="uk-card-header">
        <div className="uk-grid-small uk-flex-middle" uk-grid="">
          <div className="uk-width-auto">
            <img
              className="uk-border-circle"
              width="40"
              height="40"
              src={card.profile.img}
              alt="user-pic"
            />
          </div>
          <div className="uk-width-expand">
            <h3 className="uk-card-title uk-margin-remove-bottom">
              <NavLink to={`/card/${card.id}`}>{card.title}</NavLink>
              <EditCard cardAuthorId={card.profile.id} cardId={card.id} />
            </h3>
            <div className="uk-flex uk-flex-between">
              <p className="uk-text-meta uk-margin-remove-top uk-margin-remove-bottom">
                {card.profile.name}
              </p>
              <span className="uk-text-meta">{card.date}</span>
            </div>
          </div>
        </div>
      </div>
      <div className="uk-card-body">
        <p>
          {card.text.length > 200
            ? card.text.substring(0, 200) + "..."
            : card.text}
        </p>
      </div>
    </div>
  );
};
