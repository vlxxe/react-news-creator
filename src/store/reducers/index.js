import { combineReducers } from "redux";
import { googleAuthReducer } from "./googleAuthReducer";
import { cardsReducer } from "./cardsReducer";
import { reducer as formReducer } from "redux-form";

export const rootReducer = combineReducers({
  googleAuth: googleAuthReducer,
  cards: cardsReducer,
  form: formReducer
});
