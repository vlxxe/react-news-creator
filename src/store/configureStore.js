import { createStore, applyMiddleware } from "redux"
import { rootReducer } from "./reducers"
import logger from "redux-logger"
import thunk from "react-thunk"

export const store = createStore(rootReducer, applyMiddleware(logger))
