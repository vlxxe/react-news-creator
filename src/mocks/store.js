import configureStore from "redux-mock-store";
import thunk from "redux-thunk";

const mockStore = configureStore([thunk]);

export let emptyStore = mockStore({
  googleAuth: {
    userProfile: {
      id: null,
      name: null,
      img: null
    }
  },
  cards: {
    cardsList: [],
    cardContent: null,
    isLoading: false
  }
});
