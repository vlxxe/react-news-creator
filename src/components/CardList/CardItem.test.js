import React from "react";
import { Provider } from "react-redux";
import { BrowserRouter as Router } from "react-router-dom";
import renderer from "react-test-renderer";

import { CardItem } from "./CardItem";
import { emptyStore } from "../../mocks/store";

describe("CardItem", () => {
  let component;
  let card = {
    profile: {
      img: "http"
    },
    text: "text"
  };
  beforeEach(() => {
    component = renderer.create(
      <Provider store={emptyStore}>
        <Router>
          <CardItem card={card} />
        </Router>
      </Provider>
    );
  });

  it("should match the snapshot", () => {
    expect(component.toJSON()).toMatchSnapshot();
  });
});
