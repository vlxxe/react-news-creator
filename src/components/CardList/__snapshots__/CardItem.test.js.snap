// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardItem should match the snapshot 1`] = `
<div
  className="uk-card uk-card-default"
  style={
    Object {
      "marginBottom": 15,
      "width": 580,
    }
  }
>
  <div
    className="uk-card-header"
  >
    <div
      className="uk-grid-small uk-flex-middle"
      uk-grid=""
    >
      <div
        className="uk-width-auto"
      >
        <img
          alt="user-pic"
          className="uk-border-circle"
          height="40"
          src="http"
          width="40"
        />
      </div>
      <div
        className="uk-width-expand"
      >
        <h3
          className="uk-card-title uk-margin-remove-bottom"
        >
          <a
            aria-current={null}
            href="/card/undefined"
            onClick={[Function]}
          />
          <div
            style={
              Object {
                "position": "absolute",
                "right": 15,
                "top": 5,
              }
            }
          >
            <span
              className="uk-icon-link"
              style={
                Object {
                  "cursor": "pointer",
                  "marginRight": 5,
                }
              }
              uk-icon="pencil"
            />
            <span
              className="uk-icon-link"
              onClick={[Function]}
              style={
                Object {
                  "cursor": "pointer",
                }
              }
              uk-icon="trash"
            />
          </div>
        </h3>
        <div
          className="uk-flex uk-flex-between"
        >
          <p
            className="uk-text-meta uk-margin-remove-top uk-margin-remove-bottom"
          />
          <span
            className="uk-text-meta"
          />
        </div>
      </div>
    </div>
  </div>
  <div
    className="uk-card-body"
  >
    <p>
      text
      ...
    </p>
  </div>
</div>
`;
